### ISSUE TITLE
system clipboard integration with newlines and  other characters

### ISSUE THREAD

**Issue by nonumeros** [mawww/kakoune#4497]
### Question

I have a `map -docstring 'case insensitive search' global user '/' /(?i)` on the initialization file

But if I were to have a buffer with say a newline ␤ followed by say `//`, and another newline ␤ 

and I were to select up to the last `/` then it follows that if I were to paste the contents of the system clipboard it would return `(?i)/`, correct?

Also if I were to have a buffer with say a newline ␤ followed by `u/a`, and another newline ␤ 

But out of curiosity, what exactly is happening if I were to select only to the `/` or 

Or say for example I have a ␤ followed by `u/a` and were to select up to the `a`, then what happened to my `u` as soon as the contents from the clipboard are pasted on the minibuffer, by say e.g. with a write operation `w`

In other words, the only viable solution to retain the contents of such clipboard, such as ␤ `u/a` would be selecting also a ␤  afterwards or else, the remainder is lost
